### PR TITLE
DomainFileStore and Folder Dialogs

### PR DESCRIPTION
**Pull request by lxatrhea**
- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
- DomainFileStore button is visible again in Model tab.
- Add / Edit / Remove DomainFileStore is now possible.
- Add / Remove Folder is now possible, but only for folders directly under DomainFileStore. Not the nested Folders.
- Added unit tests for new files, not for existing (separate issue).

### CODE REVIEW

File: EngineeringModel.Tests/Dialogs/DomainFileStoreDialogViewModelTestFixture.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
+﻿// -------------------------------------------------------------------------------------------------
+// <copyright file="DomainFileStoreDialogViewModelTestFixture.cs" company="RHEA System S.A.">
```
Comment (samatrhea) [inline EngineeringModel.Tests/Dialogs/DomainFileStoreDialogViewModelTestFixture.cs]: please use AGPL license header on all new and touched files

```
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="DomainFileStoreDialogViewModelTestFixture.cs" company="RHEA System S.A.">
//    Copyright (c) 2015-2020 RHEA System S.A.
//
//    Author: Sam Gerené, Alex Vorobiev, Naron Phou, Patxi Ozkoidi, Alexander van Delft,
//
//    This file is part of CDP4-IME Community Edition. 
//    The CDP4-IME Community Edition is the RHEA Concurrent Design Desktop Application and Excel Integration
//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
//
//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
//    modify it under the terms of the GNU Affero General Public
//    License as published by the Free Software Foundation; either
//    version 3 of the License, or any later version.
//
//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
//    but WITHOUT ANY WARRANTY; without even the implied warranty of
//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
//    GNU Affero General Public License for more details.
//
//    You should have received a copy of the GNU Affero General Public License
//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
// </copyright>
// --------------------------------------------------------------------------------------------------------------------
```

**Comment by mmilitaru** [comment]
Some of the lines of code are very long and hard to follow. Please spread them to multiple lines.

**Comment by samatrhea** [comment]
code looks good, i have 3 failing tests though:
  - DomainFileStoreBrowserRibbonViewModelTestFixture.VerifyThatInstantiatePanelViewModelReturnsExpectedViewModel
  - DomainFileStoreBrowserViewModelTestFixture.VerifyThatRowsAreCreatedAndAddedCorrectly
  - DomainFileStoreBrowserViewModelTestFixture.VerifyUploadFileCommand

can you change the visibility of class `DomainFileStoreBrowserViewModelTestFixture`

File: EngineeringModel.Tests/Dialogs/DomainFileStoreDialogViewModelTestFixture.cs (hunk 1 above)
Comment (lxatrhea) [reply EngineeringModel.Tests/Dialogs/DomainFileStoreDialogViewModelTestFixture.cs]: I see Patxi here but not Merlin.
Who need to be there and who not?

Comment (samatrhea) [reply EngineeringModel.Tests/Dialogs/DomainFileStoreDialogViewModelTestFixture.cs]: please add Merlin Bieze, Mihail, Kamil and Nathaniel
